File: frontend/src/types/transfer.ts
export type TransferStatus = 'INITIATED' | 'OFFICER_VERIFIED' | 'ADMIN_APPROVED' | 'REJECTED' | 'CANCELLED';

export interface Transfer {
  id: number;
  landRecordId: number;
  kittaNumber: string;
  sellerId: number;
  sellerName: string;
  buyerId: number;
  buyerName: string;
  status: TransferStatus;
  initiatedAt: string;
  officerVerifiedAt: string | null;
  verifiedByOfficerName: string | null;
  adminApprovedAt: string | null;
  approvedByAdminName: string | null;
  rejectionReason: string | null;
  oldRecordHash: string | null;
  newRecordHash: string | null;
}

export interface TransferRequest {
  landRecordId: number;
  buyerId: number;
}

File: backend/land-record-management-backend/src/main/java/np/com/abhishekojha/landrecordmanagementbackend/model/entity/Transfer.java
package np.com.abhishekojha.landrecordmanagementbackend.model.entity;

import jakarta.persistence.*;
import lombok.*;
import np.com.abhishekojha.landrecordmanagementbackend.model.enums.TransferStatus;

import java.time.LocalDateTime;

@Entity
@Table(name = "transfers")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Transfer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "land_record_id", nullable = false)
    private LandRecord landRecord;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "seller_id", nullable = false)
    private User seller;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "buyer_id", nullable = false)
    private User buyer;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private TransferStatus status = TransferStatus.INITIATED;

    @Column(name = "initiated_at", nullable = false)
    @Builder.Default
    private LocalDateTime initiatedAt = LocalDateTime.now();

    @Column(name = "officer_verified_at")
    private LocalDateTime officerVerifiedAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "verified_by_officer_id")
    private User verifiedByOfficer;

    @Column(name = "admin_approved_at")
    private LocalDateTime adminApprovedAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "approved_by_admin_id")
    private User approvedByAdmin;

    @Column(name = "rejection_reason")
    private String rejectionReason;

    @Column(name = "old_record_hash")
    private String oldRecordHash;

    @Column(name = "new_record_hash")
    private String newRecordHash;
}
